ui/src/hooks/useKeyboard.ts

```diff
@@ -19,17 +20,36 @@ export default function useKeyboard() {
   // dynamically set when the device responds to the first key press event or reports its
   // keysDownState when queried since the keyPressReport was introduced together with the 
   // getKeysDownState API.
-  const { keyPressReportApiAvailable, setkeyPressReportApiAvailable} = useHidStore();
+  const { keyPressReportApiAvailable, setkeyPressReportApiAvailable } = useHidStore();
+
+  // HidRPC is a binary format for exchanging keyboard and mouse events
+  const { reportKeyboardEvent, reportKeypressEvent, rpcHidReady } = useHidRpc((message) => {
+    if (message.type === HID_RPC_MESSAGE_TYPES.KeysDownState) {
+      if (!message.keysDownState) {
+        return;
+      }
+
+      setKeysDownState(message.keysDownState);
+      setkeyPressReportApiAvailable(true);
+    }
+  });
 
   // sendKeyboardEvent is used to send the full keyboard state to the device for macro handling
   // and resetting keyboard state. It sends the keys currently pressed and the modifier state.
   // The device will respond with the keysDownState if it supports the keyPressReport API
   // or just accept the state if it does not support (returning no result)
   const sendKeyboardEvent = useCallback(
     async (state: KeysDownState) => {
-      if (rpcDataChannel?.readyState !== "open") return;
+      if (rpcDataChannel?.readyState !== "open" && !rpcHidReady) return;
```

Comment:
`&&` or `||`? Seems like we wanna skip it if _either_ is not ready? Or is the idea that if the old-school (`send` method) is not ready **nor** is the new-wave distinct RPC channel ready...

Reply:
It is skipped only if both RPC methods are unavailable, so use `&&` here.